src/Symfony/Component/Messenger/DependencyInjection/MessengerPass.php

```diff
@@ -165,9 +165,13 @@ private function registerHandlers(ContainerBuilder $container, array $busIds): v
         $container->addDefinitions($definitions);
 
         foreach ($busIds as $bus) {
-            $container->register($locatorId = $bus.'.messenger.handlers_locator', HandlersLocator::class)
-                ->setArgument(0, $handlersLocatorMappingByBus[$bus] ?? [])
-            ;
+            $locatorId = $bus.'.messenger.handlers_locator';
+            if (!$container->has($locatorId)) {
+                $container->register($locatorId, HandlersLocator::class)
```

Comment:
Could it be interesting to give the handlers locator to the existing definition?
```suggestion
            ($container->hasDefinition($locatorId) ? $container->getDefinition($locatorId) : $container->register($locatorId, HandlersLocator::class))
```